File: integrations/aspnetcore/src/Scalar.AspNetCore/Enums/ScalarTheme.cs
using System.ComponentModel;
using NetEscapades.EnumGenerators;

namespace Scalar.AspNetCore;

/// <summary>
/// Represents the different themes available in Scalar.
/// </summary>
[EnumExtensions]
public enum ScalarTheme
{
    /// <summary>
    /// No theme applied.
    /// </summary>
    [Description("none")]
    None,

    /// <summary>
    /// Alternate theme.
    /// </summary>
    [Description("alternate")]
    Alternate,

    /// <summary>
    /// Default theme.
    /// </summary>
    [Description("default")]
    Default,

    /// <summary>
    /// Moon theme.
    /// </summary>
    [Description("moon")]
    Moon,

    /// <summary>
    /// Purple theme.
    /// </summary>
    [Description("purple")]
    Purple,

    /// <summary>
    /// Solarized theme.
    /// </summary>
    [Description("solarized")]
    Solarized,

    /// <summary>
    /// Blue Planet theme.
    /// </summary>
    [Description("bluePlanet")]
    BluePlanet,

    /// <summary>
    /// Saturn theme.
    /// </summary>
    [Description("saturn")]
    Saturn,

    /// <summary>
    /// Kepler theme.
    /// </summary>
    [Description("kepler")]
    Kepler,

    /// <summary>
    /// Mars theme.
    /// </summary>
    [Description("mars")]
    Mars,

    /// <summary>
    /// Deep Space theme.
    /// </summary>
    [Description("deepSpace")]
    DeepSpace,
    
    /// <summary>
    /// Laserwave theme.
    /// </summary>
    [Description("laserwave")]
    Laserwave
}

File: integrations/java/webjar/src/main/java/com/scalar/maven/webjar/enums/ScalarTheme.java
package com.scalar.maven.webjar.enums;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

/**
 * Represents the different themes available in Scalar.
 */
public enum ScalarTheme {
    /**
     * No theme applied.
     */
    NONE("none"),

    /**
     * Alternate theme.
     */
    ALTERNATE("alternate"),

    /**
     * Default theme.
     */
    DEFAULT("default"),

    /**
     * Moon theme.
     */
    MOON("moon"),

    /**
     * Purple theme.
     */
    PURPLE("purple"),

    /**
     * Solarized theme.
     */
    SOLARIZED("solarized"),

    /**
     * Blue Planet theme.
     */
    BLUE_PLANET("bluePlanet"),

    /**
     * Saturn theme.
     */
    SATURN("saturn"),

    /**
     * Kepler theme.
     */
    KEPLER("kepler"),

    /**
     * Mars theme.
     */
    MARS("mars"),

    /**
     * Deep Space theme.
     */
    DEEP_SPACE("deepSpace"),

    /**
     * Laserwave theme.
     */
    LASERWAVE("laserwave");

    private final String value;

    ScalarTheme(String value) {
        this.value = value;
    }

    /**
     * Creates a ScalarTheme from a string value.
     *
     * @param value the string value
     * @return the corresponding ScalarTheme
     * @throws IllegalArgumentException if the value is not recognized
     */
    @JsonCreator
    public static ScalarTheme fromValue(String value) {
        for (ScalarTheme theme : values()) {
            if (theme.value.equals(value)) {
                return theme;
            }
        }
        throw new IllegalArgumentException("Unknown theme: " + value);
    }

    /**
     * Gets the string value for JSON serialization.
     *
     * @return the string value
     */
    @JsonValue
    public String getValue() {
        return value;
    }
}
